minigeth/core/state_processor.go

```diff
@@ -84,75 +88,120 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	// check batch signature
 
 	// check decryption key against eon key in contract
-	decryptionKey := contextTx.DecryptionKey()
-	fmt.Println(decryptionKey)
-
-	// execute envelopes of encrypted txs. Keep track of nonces to be used later for execution the payloads
-	feePaid := make(map[int]bool)
-	coinbase := block.Coinbase()
-	for i, tx := range block.Transactions() {
-		feePaid[i] = false
-		if tx.Type() != types.ShutterTxType {
-			continue
-		}
-		sender, err := signer.Sender(tx)
-		if err != nil {
-			return nil, nil, 0, fmt.Errorf("could not extract signer of tx %d [%v]: %w", i, tx.Hash().Hex(), err)
-		}
-		gasPrice := math.BigMin(new(big.Int).Add(tx.GasTipCap(), header.BaseFee), tx.GasFeeCap())
-		gasFee := new(big.Int).Mul(gasPrice, new(big.Int).SetUint64(tx.Gas()))
-
-		balance := statedb.GetBalance(sender)
-		if balance.Cmp(gasFee) > 0 {
-			statedb.SubBalance(sender, gasFee)
-			statedb.AddBalance(coinbase, gasFee)
-			feePaid[i] = true
-		}
-	}
+	if p.config.EonKeyBroadcastAddress.String() != "" {
+		decryptionKey := contextTx.DecryptionKey()
+		fmt.Println(decryptionKey)
+
+		blankTxContext := vm.TxContext{Origin: common.Address{}, GasPrice: common.Big0}
+		e := vm.NewEVM(blockContext, blankTxContext, statedb, p.config, cfg)
+		contract := vm.AccountRef(p.config.EonKeyBroadcastAddress)
+
+		selector := crypto.Keccak256([]byte("get(uint64)"))[:4]
+		paddedBlk := common.BigToHash(blockNumber)
+		callData := append(selector, paddedBlk.Bytes()...)
 
-	// decrypt and execute payloads of encrypted txs if they paid the transaction fee
-	// if this fails, at least increment the sender's nonce, so that the tx can't be replayed
-	for i, tx := range block.Transactions() {
-		if tx.Type() != types.ShutterTxType {
-			continue
+		result, remainingGas, err := e.Call(contract, p.config.EonKeyBroadcastAddress, callData, 1000000, common.Big0)
+		if (remainingGas == 0) {
+			return nil, nil, 0, fmt.Errorf("call on eon key storage ran out of gas")
 		}
-		if !feePaid[i] {
-			continue
+		if (err != nil) {
+			log.Printf("could not find an eon key for block number %s", blockNumber)
+			result = []byte{}
 		}
 
-		sender, err := signer.Sender(tx)
-		if err != nil {
-			return nil, nil, 0, fmt.Errorf("could not extract signer of tx %d [%v]: %w", i, tx.Hash().Hex(), err)
+		// decode result with abi
+		eonKeyBytes := []byte{}
+		if bytes.Compare(result, []byte{}) != 0 {
```

Comment:
`bytes.Equal` or just `len(result) != 0` feels more appropriate (since we don't care about which one's "bigger"